openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -1340,6 +1340,7 @@ This specification defines the following Credential Issuer Metadata:
   * `enc_values_supported`: REQUIRED. Array containing a list of the JWE [@!RFC7516] encryption algorithms (`enc` values) [@!RFC7518] supported by the Credential and Batch Credential Endpoint to encode the Credential or Batch Credential Response in a JWT [@!RFC7519].
   * `encryption_required`: REQUIRED. Boolean value specifying whether the Credential Issuer requires the additional encryption on top of TLS for the Credential Response. If the value is `true`, the Credential Issuer requires the encryption for every Credential Response and therefore the Wallet MUST provide encryption keys in the Credential Request. If the value is `false`, the Wallet MAY chose whether it provides encryption keys or not.
 * `credential_identifiers_supported`: OPTIONAL. Boolean value specifying whether the Credential Issuer supports returning `credential_identifiers` parameter in the `authorization_details` Token Response parameter, with `true` indicating support. If omitted, the default value is `false`.
+* `signed_metadata`: OPTIONAL. String that is a signed JWT. This JWT contains Credential Issuer metadata parameters as claims. The signed metadata MUST be secured using JSON Web Signature (JWS) [@!RFC7515] and MUST contain an `iat` (Issued At) claim and an `iss` (Issuer) claim denoting the party attesting to the claims in the signed metadata. This JWT can also be used to identify the actual Credential Issuer when it is using third party service provider as a domain to host its metadata file, in which case, the JWT SHOULD contain `sub` (Subject) claim matching the Credentials Issuer identifier. If the Wallet supports signed metadata, metadata values conveyed in the signed JWT MUST take precedence over the corresponding values conveyed using plain JSON elements. A `signed_metadata` metadata value SHOULD NOT appear as a claim in the JWT. The Wallet MUST establish trust in the signer of the metadata, and obtain the keys to validate the signature. The concrete mechanism how to do that is out of scope of this specification and may be defined in the profiles of this specification.
```

Comment:
```suggestion
* `signed_metadata`: OPTIONAL. String that is a signed JWT. This JWT contains Credential Issuer metadata parameters as claims. The signed metadata MUST be secured using JSON Web Signature (JWS) [@!RFC7515] and MUST contain an `iat` (Issued At) claim and an `iss` (Issuer) claim denoting the party attesting to the claims in the signed metadata. This JWT can also be used to identify the actual Credential Issuer when it is using a third party service provider as a domain to host its metadata file, in which case, the JWT SHOULD contain `sub` (Subject) claim matching the Credentials Issuer identifier. If the Wallet supports signed metadata, metadata values conveyed in the signed JWT MUST take precedence over the corresponding values conveyed using plain JSON elements. If the Issuer wants to enforce use of signed metadata, it SHOULD omit the respective metadata parameters from the unsigned part of the credential issuer metadata. A `signed_metadata` metadata value SHOULD NOT appear as a claim in the JWT. The Wallet MUST establish trust in the signer of the metadata, and obtain the keys to validate the signature. The concrete mechanism how to do that is out of scope of this specification and may be defined in the profiles of this specification.
```

Reply:
I agree with @tlodderstedt , otherwise we won't have interop. I added some editorial suggestions plus saying that the `signed_metadata` MUST NOT appear in the `signed_metadata`.

```suggestion
* `signed_metadata`: OPTIONAL. The `signed_metadata` parameter is a signed JSON Web Token (JWT) containing Credential Issuer metadata parameters as claims. The `signed_metadata` parameter MUST be secured using JSON Web Signature (JWS) [@!RFC7515] and MUST include an `iat` (Issued At) claim and an `iss` (Issuer) claim, indicating the party attesting to the claims in the `signed_metadata` parameter. The `signed_metadata` parameter can serve to identify the actual Credential Issuer when using a third-party service provider as a domain to host its Credential Issuer metadata. In such cases, the `signed_metadata` parameter SHOULD include a `sub` (Subject) claim matching the Credential Issuer identifier. If the Wallet supports the `signed_metadata` parameter, the metadata parameters conveyed as claims in the `signed_metadata` parameter MUST take precedence over the corresponding metadata parameters in the unsigned embedding Credential Issuer metadata. If the Issuer wishes to enforce the use of the `signed_metadata` parameter, it MUST exclude the respective metadata parameters contained as claims in the `signed_metadata` parameter from the unsigned Credential Issuer metadata. The `signed_metadata` parameter MUST NOT contain a `signed_metadata` claim. The Wallet MUST establish trust in the signer of the JWT and obtain the keys to validate the signature before utilizing the Credential Issuer metadata parameter values conveyed as claims in the `signed_metadata` parameter value. The specific mechanism for doing so is beyond the scope of this specification and MAY be defined in the profiles associated with this specification.
```

Reply:
>  If the Issuer wants to enforce use of signed metadata, it SHOULD omit the respective metadata parameters from the unsigned part of the credential issuer metadata.

@tlodderstedt IMO, this is unnecessary since the `signed_metadata` takes precedence. What happens if a wallet cannot validate the JWS and finds the same parameter in the `signed_metadata` and the unsigned metadata? According to this spec, it MUST fail since `signed_metadata` takes precedence. Do we need to clarify this better?

Reply:
Should the unsigned metadata be a fallback in case the wallet cannot validate the JWS, or do we accept the fact that in that case wallets that don't trust the keys cannot proceed? In the latter case we should probably change the MUST to a "SHOULD establish trust".

Reply:
If the wallet supports signed metadata it must use that data only if it trusts the public key, otherwise it should use the plaintext values. So if the issuer enforces all the metadata being signed and the wallet can't establish trust in the public key transaction should fail. So current MUST is fine.

Reply:
After reading the text again, I think it should be fine. We could always be more explicit by saying something like the following but I'm not insisting:
```suggestion
* `signed_metadata`: OPTIONAL. String that is a signed JWT. This JWT contains Credential Issuer metadata parameters as claims. The signed metadata MUST be secured using JSON Web Signature (JWS) [@!RFC7515] and MUST contain an `iat` (Issued At) claim and an `iss` (Issuer) claim denoting the party attesting to the claims in the signed metadata. This JWT can also be used to identify the actual Credential Issuer when it is using a third party service provider as a domain to host its metadata file, in which case, the JWT SHOULD contain `sub` (Subject) claim matching the Credentials Issuer identifier. If the Wallet supports signed metadata, metadata values conveyed in the signed JWT MUST take precedence over the corresponding values conveyed using plain JSON elements. If the Issuer wants to enforce use of signed metadata, it SHOULD omit the respective metadata parameters from the unsigned part of the credential issuer metadata. A `signed_metadata` metadata value SHOULD NOT appear as a claim in the JWT. The Wallet MUST establish trust in the signer of the metadata, and obtain the keys to validate the signature. If the Wallet cannot establish trust in the signer of the metadata, the Wallet MUST reject the metadata parameters contained in the `signed_metadata` parameter value. The concrete mechanism how to do that is out of scope of this specification and may be defined in the profiles of this specification.
```